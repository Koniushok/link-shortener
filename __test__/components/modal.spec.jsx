import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Modal from '../../src/components/modal';

describe('Modal component', () => {
  const props = {
    handelClose: () => {},
  };

  describe('Modal component initial', () => {
    const modal = shallow(<Modal {...props} />);
    it('render correctly component', () => {
      expect(shallowToJson(modal)).toMatchSnapshot();
    });
  });
  describe('Modal isLoader', () => {
    const nextProps = {
      ...props,
      loading: true,
      children: <p />,
    };
    const modal = shallow(<Modal {...nextProps} />);
    it('render only children', () => {
      expect(modal.find('p')).toHaveLength(1);
    });
  });
  describe('call handelClose', () => {
    const handelClose = jest.fn();
    const nextProps = {
      ...props,
      handelClose,
    };
    const modal = mount(<Modal {...nextProps} />);
    it('call handelClose when click <ModalWrapper />', () => {
      const modalWrapper = modal.find('ModalWrapper');
      modalWrapper.simulate('click');
      expect(handelClose).toHaveBeenCalledTimes(1);
    });
  });
  describe('not call handelClose', () => {
    const handelClose = jest.fn();
    const nextProps = {
      ...props,
      handelClose,
    };
    const modal = mount(<Modal {...nextProps} />);
    it('not call handelClose when click <ModalContent />', () => {
      const modalContent = modal.find('ModalContent');
      modalContent.simulate('click');
      expect(handelClose).toHaveBeenCalledTimes(0);
    });
  });
});
