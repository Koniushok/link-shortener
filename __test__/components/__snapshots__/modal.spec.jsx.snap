// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component Modal component initial render correctly component 1`] = `
<ModalWrapper
  onClick={[Function]}
>
  <GlobalStyleComponent />
  <ModalContent>
    <ButtonClose
      onClick={[Function]}
    >
      <Close />
    </ButtonClose>
  </ModalContent>
</ModalWrapper>
`;
